### PR TITLE
Cleanup/review

### PR DESCRIPTION
**Pull request by p3lim**
Misc cleanup for the 7.0 release, push anything you find to this branch.

### CODE REVIEW

File: elements/additionalpower.lua [hunk 1]
```diff
@@ -98,10 +100,10 @@ local function Update(self, event, unit, powertype)
 	--[[ Callback: AdditionalPower:PostUpdate(unit, cur, max)
 	Called after the element has been updated.
 
-	* self - the AdditionalPower element
-	* unit - the event unit that the update has been triggered for
-	* cur  - the current value of the player's additional power
-	* max  - the maximum possible value of the player's additional power
+	* self - AdditionalPower element
+	* unit - the unit for which the update has been triggered
```
Comment (Rainrider) [inline elements/additionalpower.lua]: ha! you forgot a `the` here :)

File: elements/additionalpower.lua [hunk 2]
```diff
@@ -1,31 +1,32 @@
...
 # Element: Additional Power Bar
 
-Handles updating and visibility of a status bar displaying the player's additional power, such as Mana for Balance
+Handles updating and visibility of a status bar that displays player's additional power, such as Mana for Balance
 druids.
 
 ## Widget
 
-AdditionalPower - A StatusBar to represent the player's current additional power.
+AdditionalPower - A StatusBar that is used to represent player's additional power.
```
Comment (p3lim) [inline elements/additionalpower.lua]: "A StatusBar that is used to display the player's additional power."

File: elements/additionalpower.lua [hunk 3]
```diff
@@ -1,31 +1,32 @@
...
 
 ## Sub-Widgets
 
-.bg - A Texture, which functions as a background. It will inherit the color of the main StatusBar.
+.bg - A Texture that is used as a background. Inherits the widget's color.
```
Comment (p3lim) [inline elements/additionalpower.lua]: element, not widget.

File: elements/additionalpower.lua [hunk 4]
```diff
@@ -1,31 +1,32 @@
...
 
 ## Notes
 
-The default StatusBar texture will be applied if the UI widget doesn't have a status bar texture or color defined.
+Default status bar texture will be applied if the widget doesn't have a texture or a color set.
```
Comment (p3lim) [inline elements/additionalpower.lua]: "The default status bar texture will be applied if the element doesn't have a texture or a color set."

File: elements/additionalpower.lua [hunk 5]
```diff
@@ -1,31 +1,32 @@
...
 
 ## Options
 
-.colorClass  - Use `self.colors.class[class]` to color the bar.
-.colorSmooth - Use `self.colors.smooth` to color the bar with a smooth gradient based on the players current additional
+.colorClass  - Uses `self.colors.class[class]` to color the bar based on player's class.
+.colorSmooth - Uses `self.colors.smooth` to color the bar with a smooth gradient based on player's current additional
```
Comment (p3lim) [inline elements/additionalpower.lua]: **The** player val.

Comment (p3lim) [inline elements/additionalpower.lua]: And I don't agree with the change from "use" to "uses" on any of these.

File: elements/additionalpower.lua (hunk 2 above)
Comment (ls-) [inline elements/additionalpower.lua]: "A StatusBar that is used to display the player's additional power." would be better imho...

File: elements/additionalpower.lua (hunk 3 above)
Comment (ls-) [inline elements/additionalpower.lua]: Hm... Why? O_o Widget means status bar.

File: elements/alternativepower.lua [hunk 6]
```diff
@@ -159,7 +158,7 @@ local function Enable(self, unit)
 			end
 
 			--[[ Override: AlternativePower:UpdateTooltip()
-			Called when the widget is hovered. Used to populate the tooltip.
+			Called when mouse is over the widget. Used to populate its tooltip.
```
Comment (p3lim) [inline elements/alternativepower.lua]: the

File: elements/alternativepower.lua [hunk 7]
```diff
@@ -1,18 +1,17 @@
...
-Handles visibility and updating of the alternative power bar.
-This bar is used to display encounter/quest related power information, such as the number of hour glass uses left on the
-end boss in End Time.
+Handles updating and visibility of a status bar that displays encounter- or quest-related power information, such as the number of hour glass
+charges during Murozond encounter in End Time.
 
 ## Widget
 
-AlternativePower - A StatusBar to represent alternative power.
+AlternativePower - A StatusBar that is used to represent the unit's alternative power.
```
Comment (p3lim) [inline elements/alternativepower.lua]: drop "that is"

File: elements/alternativepower.lua [hunk 8]
```diff
@@ -1,18 +1,17 @@
...
 
 ## Notes
 
-`OnEnter` and `OnLeave` handlers to display a tooltip will be set on the widget if it is mouse enabled.
-The default StatusBar texture will be applied if the UI widget doesn't have a status bar texture or color defined.
+If mouse interactivity is enabled for the widget, `OnEnter` and/or `OnLeave` handlers will be set to display a tooltip.
+Default status bar texture will be applied if the widget doesn't have a texture or a color set.
```
Comment (Rainrider) [inline elements/alternativepower.lua]: __A__ default or __The__ default imho

File: elements/assistantindicator.lua [hunk 9]
```diff
@@ -1,15 +1,15 @@
...
+Toggles visibility of an indicator based on the unit's raid assistant status.
 
 ## Widget
 
 AssistantIndicator - Any UI widget.
 
 ## Notes
 
-The default assistant icon will be applied if the UI widget is a texture and doesn't have a texture or color defined.
+Default assistant icon will be applied if the widget is a Texture and doesn't have a texture or a color set.
```
Comment (Rainrider) [inline elements/assistantindicator.lua]: __The__ default. There is only one default, isn't it?

**Comment by p3lim** [comment]
Some notes:
- Castbar has no overrides
- Range has no ForceUpdate method

We also have to decide on the type notation.  
In some cases the documentation prefixes the description of a parameter with the type, e.g. "a Boolean" or "a String", while in others it doesn't.

We need to decide if we want to prefix the descriptions like this or not, and if we do, what format we want to prefix them with ("a Boolean", "boolean value" etc).

**Comment by p3lim** [comment]
- Any reason we arent using the following constants in blizzard.lua?
```
MAX_ARENA_ENEMIES = 5
MAX_BOSS_FRAMES = 5
MAX_PARTY_MEMBERS = 4
```
- The following line doesn't allow for color overriding per-layout, have to override the global colors to change it:
https://github.com/oUF-wow/oUF/blob/development/elements/classpower.lua#L46
- Totems are not hidden while in a vehicle

**Comment by ls-** [comment]
> We need to decide if we want to prefix the descriptions like this or not, and if we do, what format we want to prefix them with ("a Boolean", "boolean value" etc).

IMHO, if we expect to receive parameters of certain types, we usually call `argcheck` to test them, so I think we should mention expected types in such cases, otherwise it's unnecessary.

Moreover, we may want to add `argcheck`s where they're needed, and ofc remove redundant ones.

**Comment by ls-** [comment]
>Totems are not hidden while in a vehicle.
Totems aren't only totems, various CDs, familiars and other things are counted as totems. For instance, DK's Ebon Gargoyle is shown as a "totem".

So theoretically some vehicle may have "totems" too.

**Comment by p3lim** [comment]
> So theoretically some vehicle may have "totems" too.

Has never existed such a thing afaik.

**Comment by Rainrider** [comment]
What about appending the type at the end of the description? Like: `* name - name of the element (string)`
This way the type won't lead to any change in wording or need special highlighting. Lua type could be written in small (i.e. `boolean`), WoW types in PascalCase (i.e. `StatusBar` or `Texture`) (I'd expect anybody reading the docs can keep them apart though). Specifying the type is nice, so people know what to expect.

**Comment by p3lim** [comment]
@Rainrider Good suggestion, and I agree on everything. Singling out widget types is nice, as it's more relevant than variable types.

**Comment by ls-** [comment]
> Has never existed such a thing afaik.

Blizz constantly add/remove abilities to/from "totems", e.g. Mage's Arcane Familiar was a "totem" at some point, now it's just a buff, DK's secondary minion (archer) used to be a "totem" too, but is tied to main summon now.

IMHO, we should leave them as is.

**Comment by ls-** [comment]
@p3lim checked Blizz totem code, although it's shit, they do [this](https://github.com/Gethe/wow-ui-source/blob/live/FrameXML/TotemFrame.lua#L53-L55). So yeah, even theoretically that case is kinda  impossible, guess we can hide it.

**Comment by Rainrider** [comment]
@ls- the code you linked always shows totem timers for shamans and even calculates their placement based on pet presence for DK, so totems actually don't disappear in vehicles.

**Comment by p3lim** [comment]
@Rainrider But the unitframe they are anchored to does. In oUF's case we just swap units on the unitframes, not swapping the entire frames, thus Totems are still connected to the player frame.

**Comment by ls-** [comment]
@Rainrider that's not what I tried to point out :p If `PetFrame` is shown and player's class isn't one of those 6, they just hide `TotemFrame` and prevent it from updating.

@p3lim Default UI does exactly the same thing, when a player gets inside a vehicle `PlayerFrame` shows unit and player is shown as `PetFrame`.

**Pull request merged**